### DISCUSSION TITLE
Data Type When Writing

### DISCUSSION THREAD

**Discussion by pmojo375** [ottowayi/pycomm3#155] [Q&A]
I don't have access to a PLC to test this on right now or I would check myself but I am writing some code to read/write tags and am curious if the library can take tag values as strings in the write function when the PLC tag is an INT, DINT, REAL, etc. or if I need to have the type match up to the PLC tag type I am writing? I see that you had highlighted the simplified data types as a feature but wasn't sure if this was only for the more complex reasons that were stated or not. For example, to write to a DINT type in the PLC, would I need to ensure that the value I pass to the function is an int or can I pass a string (what my code currently would try to do) and have it handle that internally?

**Comment by ottowayi** [comment]
Simplified data types mostly refers to strings and BOOL arrays, if you've worked with other libraries those usually have to be handled differently than all other tags.  pycomm3 will automatically convert the string structure tags (.LEN and .DATA) to/from a python `str` object sand same with BOOL arrays as lists of python `bool`s.   As well struct tags (TIMER, PID, etc, UDTs, AOIs) being just Python `dict`s.   

So if you're trying `plc.write('my_int_tag', '100')` will not work since it would not try to convert the string to an int.  I hadn't considered that as a feature before, it's possible that it could be done but I'd have to think about it more and what side effects it could lead to.   I'm not ruling it out as a possibility, but haven't looked into it enough yet to make a final decision.
